### PR TITLE
Add question generation example

### PR DESCRIPTION
**Pull request by westfish**
### PR types

New features

### PR changes

Others

### Description

Add T5-based question generation example

### CODE REVIEW

File: examples/question_generation/t5_qg/README.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+├── requirements.txt # 环境依赖文件
+└── README.md # 文档说明
+```
+
+### 数据准备
+
+#### 数据加载
+**SQuAD**数据集是一个英文问答数据集，现有的问题生成研究主要在该数据集上进行评价。**SQuAD**中的数据由段落、问题、答案3个主要部分组成，其中段落和问题从维基百科中获取，答案由人工标注。
+
+为了方便用户快速测试，PaddleNLP Dataset API内置了Squad数据集，一键即可完成数据集加载，示例代码如下：
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md:37]: Squad -> SQuAD

File: examples/question_generation/t5_qg/README.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
+# T5
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md]: 标题直接用 T5 是出于什么考虑？建议标题还是用 问题生成(Question Generation)。

File: examples/question_generation/t5_qg/README.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+## 简介
+
+Question Generation（QG），即问题生成，指的是给定一段上下文（passage或sentence），自动生成一个流畅且符合上下文主题的问句。问题生成通常可以分为两个分支，即无答案问题生成和有答案问题生成。
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md]: 无答案问题生成、有答案问题生成这些中文翻译比较容易引起误解，建议在中文术语后括号备注把对应的英文术语明确写出来。

File: examples/question_generation/t5_qg/README.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+├── generate.py # 模型生成主程序入口
+├── utils.py # 定义参数及一些工具函数
+├── requirements.txt # 环境依赖文件
+└── README.md # 文档说明
+```
+
+### 数据准备
+
+#### 数据加载
+**SQuAD**数据集是一个英文问答数据集，现有的问题生成研究主要在该数据集上进行评价。**SQuAD**中的数据由段落、问题、答案3个主要部分组成，其中段落和问题从维基百科中获取，答案由人工标注。
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md]: 注意下表述严谨: SQuAD 数据集中的 `问题` 是通过众包方式人工标注的。

File: examples/question_generation/t5_qg/README.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+```python
+from paddlenlp.datasets import load_dataset
+train_set, dev_set, test_set = load_dataset("squad",  splits=["train_v1", "dev_v1"])
+```
+
+#### 数据处理
+针对**SQuAD**数据集，我们需要将QA任务格式的数据进行转换从而得到text2text形式的数据
+```text
+answer: {answer_text} context: {context_text}
+question: {question_text}
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md:48]: 这里能否给出一个转换后的具体实例数据？帮助用户更容易理解。

File: examples/question_generation/t5_qg/README.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+- `seed` 表示随机数生成器的种子。
+
+- `ignore_pad_token_for_loss` 表示训练时计算loss时忽略padding。如果训练时设置为True，那么预测时的label需要还原来计算评估指标。
+
+- `logging_steps` 表示日志打印间隔。
+
+- `device` 表示使用的设备。
+
+程序运行结束后会将预测生成的摘要保存在`output_path`中。同时终端中会输出评估结果。
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md]: 预测生成的摘要？ 文档是作为用户入口最先看到的东西，值得用心打磨，对于推广 Question Generation 也有帮助。

File: examples/question_generation/t5_qg/README.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+采用社区微调模型mrm8488/t5-base-finetuned-question-generation-ap在验证集上有如下结果：
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/README.md:198]: 按照我们自己训练脚本训出来的模型和社区模型相比指标如何有做过评估么？

File: examples/question_generation/t5_qg/generate.py [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+        type=str,
+        choices=["cpu", "gpu", "xpu"],
+        help="The device to select to train the model, is must be cpu/gpu/xpu.")
+    parser.add_argument(
+        "--ignore_pad_token_for_loss",
+        default=True,
+        type=bool,
+        help="Whether to ignore the tokens corresponding to "
+        "padded labels in the loss computation or not.",
+    )
```
Comment (tianxin1860) [inline examples/question_generation/t5_qg/generate.py]: 预测阶段需要这个参数么？

**Review by tianxin1860 — COMMENTED** [review]
Leave some comments
